### ISSUE TITLE
Does not work with mongo version 4.4.0

### ISSUE THREAD

**Issue by saniales** [tryvium-travels/memongo#1]
Following issue on original repo: https://github.com/benweissmann/memongo/issues/8

@sbansal7

**Comment by saniales** [comment]
@sbansal7 can you please share the logs of your builds? because I am currently having a similar issue where the build fails on ubuntu-latest and working on macos-latest

**Comment by sbansal7** [comment]
https://github.com/tryvium-travels/memongo/pull/3

**Comment by sbansal7** [comment]
Here are the logs from my run: 
``` 
Run ./scripts/runUnitTests.sh
  ./scripts/runUnitTests.sh
  shell: /usr/bin/bash -e {0}
  env:
    GOROOT: /opt/hostedtoolcache/go/1.13.15/x64
go: downloading github.com/spf13/afero v1.2.1
go: downloading github.com/stretchr/testify v1.3.0
go: downloading github.com/acobaugh/osrelease v0.0.0-20181218015638-a93a0a55a249
go: downloading go.mongodb.org/mongo-driver v1.0.3
go: finding github.com/benweissmann/memongo v0.1.1
go: extracting github.com/stretchr/testify v1.3.0
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading github.com/davecgh/go-spew v1.1.0
go: extracting github.com/pmezard/go-difflib v1.0.0
go: extracting github.com/spf13/afero v1.2.1
go: downloading golang.org/x/text v0.3.0
go: downloading github.com/benweissmann/memongo v0.1.1
go: extracting github.com/davecgh/go-spew v1.1.0
go: extracting github.com/acobaugh/osrelease v0.0.0-20181218015638-a93a0a55a249
go: extracting go.mongodb.org/mongo-driver v1.0.3
go: downloading github.com/go-stack/stack v1.8.0
go: downloading golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/golang/snappy v0.0.1
go: extracting github.com/go-stack/stack v1.8.0
go: extracting golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: extracting github.com/golang/snappy v0.0.1
go: extracting github.com/xdg/stringprep v1.0.0
go: extracting github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading golang.org/x/crypto v0.0.0-20190618222545-ea8f1a30c443
go: extracting github.com/benweissmann/memongo v0.1.1
go: extracting golang.org/x/text v0.3.0
go: extracting golang.org/x/crypto v0.0.0-20190618222545-ea8f1a30c443
go: finding github.com/acobaugh/osrelease v0.0.0-20181218015638-a93a0a55a249
go: finding github.com/spf13/afero v1.2.1
go: finding golang.org/x/text v0.3.0
go: finding github.com/stretchr/testify v1.3.0
go: finding go.mongodb.org/mongo-driver v1.0.3
go: finding github.com/davecgh/go-spew v1.1.0
go: finding github.com/pmezard/go-difflib v1.0.0
go: finding github.com/go-stack/stack v1.8.0
go: finding github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: finding github.com/xdg/stringprep v1.0.0
go: finding golang.org/x/crypto v0.0.0-20190618222545-ea8f1a30c443
go: finding golang.org/x/sync v0.0.0-20190423024810-112230192c58
go: finding github.com/golang/snappy v0.0.1
[memongo] [INFO]  Starting MongoDB with options &memongo.Options{Port:33399, CachePath:"/home/runner/.cache/memongo", MongoVersion:"3.2.22", DownloadURL:"https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1604-3.2.22.tgz", MongodBin:"", Logger:(*log.Logger)(nil), LogLevel:2, StartupTimeout:10000000000}
[memongo] [INFO]  mongod from https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1604-3.2.22.tgz does not exist in cache, downloading to /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_2_22_tgz_864620453a/mongod
[memongo] [INFO]  finished downloading mongod to /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_2_22_tgz_864620453a/mongod in 43.931485885s
[memongo] [DEBUG] Using binary /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_2_22_tgz_864620453a/mongod
[memongo] [DEBUG] Starting mongod
[memongo] [DEBUG] Started mongod; starting watcher
[memongo] [DEBUG] Started watcher; waiting for mongod to report port number
[memongo] [DEBUG] [Mongod stderr] /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_2_22_tgz_864620453a/mongod: error while loading shared libraries: libssl.so.1.0.0: cannot open shared object file: No such file or directory
[memongo] [INFO]  Starting MongoDB with options &memongo.Options{Port:33091, CachePath:"/home/runner/.cache/memongo", MongoVersion:"3.4.21", DownloadURL:"https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1604-3.4.21.tgz", MongodBin:"", Logger:(*log.Logger)(nil), LogLevel:2, StartupTimeout:10000000000}
[memongo] [INFO]  mongod from https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1604-3.4.21.tgz does not exist in cache, downloading to /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_4_21_tgz_55d0a80d3f/mongod
[memongo] [INFO]  finished downloading mongod to /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_4_21_tgz_55d0a80d3f/mongod in 49.999685607s
[memongo] [DEBUG] Using binary /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_4_21_tgz_55d0a80d3f/mongod
[memongo] [DEBUG] Starting mongod
[memongo] [DEBUG] Started mongod; starting watcher
[memongo] [DEBUG] Started watcher; waiting for mongod to report port number
[memongo] [DEBUG] [Mongod stderr] /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_4_21_tgz_55d0a80d3f/mongod: error while loading shared libraries: libssl.so.1.0.0: cannot open shared object file: No such file or directory
[memongo] [INFO]  Starting MongoDB with options &memongo.Options{Port:41457, CachePath:"/home/runner/.cache/memongo", MongoVersion:"3.6.13", DownloadURL:"https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1604-3.6.13.tgz", MongodBin:"", Logger:(*log.Logger)(nil), LogLevel:2, StartupTimeout:10000000000}
[memongo] [INFO]  mongod from https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1604-3.6.13.tgz does not exist in cache, downloading to /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_6_13_tgz_fc1c3ea886/mongod
[memongo] [INFO]  finished downloading mongod to /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_6_13_tgz_fc1c3ea886/mongod in 48.355889862s
[memongo] [DEBUG] Using binary /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_6_13_tgz_fc1c3ea886/mongod
[memongo] [DEBUG] Starting mongod
[memongo] [DEBUG] Started mongod; starting watcher
[memongo] [DEBUG] [Mongod stderr] /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1604-3_6_13_tgz_fc1c3ea886/mongod: error while loading shared libraries: libcrypto.so.1.0.0: cannot open shared object file: No such file or directory
[memongo] [DEBUG] Started watcher; waiting for mongod to report port number
[memongo] [INFO]  Starting MongoDB with options &memongo.Options{Port:0, CachePath:"/home/runner/.cache/memongo", MongoVersion:"4.0.13", DownloadURL:"https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1804-4.0.13.tgz", MongodBin:"", Logger:(*log.Logger)(nil), LogLevel:2, StartupTimeout:10000000000}
[memongo] [INFO]  mongod from https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1804-4.0.13.tgz does not exist in cache, downloading to /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1804-4_0_13_tgz_adfe9ba380/mongod
[memongo] [INFO]  finished downloading mongod to /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1804-4_0_13_tgz_adfe9ba380/mongod in 38.381696526s
[memongo] [DEBUG] Using binary /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1804-4_0_13_tgz_adfe9ba380/mongod
[memongo] [DEBUG] Starting mongod
[memongo] [DEBUG] Started mongod; starting watcher
[memongo] [DEBUG] Started watcher; waiting for mongod to report port number
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.284+0000 I CONTROL  [main] Automatically disabling TLS 1.0, to force-enable TLS 1.0 specify --sslDisabledProtocols 'none'
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.286+0000 I CONTROL  [initandlisten] MongoDB starting : pid=3463 port=0 dbpath=/tmp/514651923 64-bit host=fv-az39-287
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.286+0000 I CONTROL  [initandlisten] db version v4.0.13
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.286+0000 I CONTROL  [initandlisten] git version: bda366f0b0e432ca143bc41da54d8732bd8d03c0
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.286+0000 I CONTROL  [initandlisten] OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.286+0000 I CONTROL  [initandlisten] allocator: tcmalloc
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.286+0000 I CONTROL  [initandlisten] modules: none
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.286+0000 I CONTROL  [initandlisten] build environment:
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.286+0000 I CONTROL  [initandlisten]     distmod: ubuntu1804
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.286+0000 I CONTROL  [initandlisten]     distarch: x86_64
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.286+0000 I CONTROL  [initandlisten]     target_arch: x86_64
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.286+0000 I CONTROL  [initandlisten] options: { net: { port: 0 }, storage: { dbPath: "/tmp/514651923", engine: "ephemeralForTest" } }
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] ** WARNING: Access control is not enabled for the database.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] **          Read and write access to data and configuration is unrestricted.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] ** WARNING: This server is bound to localhost.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] **          Remote systems will be unable to connect to this server. 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] **          Start the server with --bind_ip <address> to specify which IP 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] **          addresses it should serve responses from, or with --bind_ip_all to
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] **          bind to all interfaces. If this behavior is desired, start the
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] **          server with --bind_ip 127.0.0.1 to disable this warning.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] ** WARNING: /sys/kernel/mm/transparent_hugepage/enabled is 'always'.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] **        We suggest setting it to 'never'
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] ** WARNING: soft rlimits too low. rlimits set to 27785 processes, 65536 files. Number of processes should be at least 32768 : 0.5 times number of files.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] ** NOTE: The ephemeralForTest storage engine is for testing only. 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] **       Do not use in production.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I STORAGE  [initandlisten] createCollection: admin.system.version with provided UUID: 313d6e48-6e5b-48de-a55a-bd6419955c11
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.289+0000 I COMMAND  [initandlisten] setting featureCompatibilityVersion to 4.0
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.290+0000 I STORAGE  [initandlisten] createCollection: local.startup_log with generated UUID: 5e852839-57b3-45b6-bb9c-c1710e185196
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.290+0000 I FTDC     [initandlisten] Initializing full-time diagnostic data capture with directory '/tmp/514651923/diagnostic.data'
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.291+0000 I STORAGE  [LogicalSessionCacheRefresh] createCollection: config.system.sessions with generated UUID: bf43831c-dadd-4908-84f0-73929e951f38
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.291+0000 I INDEX    [LogicalSessionCacheRefresh] build index on: config.system.sessions properties: { v: 2, key: { lastUse: 1 }, name: "lsidTTLIndex", ns: "config.system.sessions", expireAfterSeconds: 1800 }
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.291+0000 I INDEX    [LogicalSessionCacheRefresh] 	 building index using bulk method; build may temporarily use up to 500 megabytes of RAM
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.291+0000 I INDEX    [LogicalSessionCacheRefresh] build index done.  scanned 0 total records. 0 secs
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.291+0000 I NETWORK  [initandlisten] waiting for connections on port 43365
[memongo] [DEBUG] mongod started up and reported a port number after 23.11186ms
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.294+0000 I NETWORK  [listener] connection accepted from 127.0.0.1:36506 #1 (1 connection now open)
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.296+0000 I NETWORK  [listener] connection accepted from 127.0.0.1:36510 #2 (2 connections now open)
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:02.296+0000 I NETWORK  [conn2] received client metadata from 127.0.0.1:36510 conn2: { driver: { name: "mongo-go-driver", version: "v1.0.3" }, os: { type: "linux", architecture: "amd64" }, platform: "go1.13.15" }
[memongo] [INFO]  Starting MongoDB with options &memongo.Options{Port:0, CachePath:"/home/runner/.cache/memongo", MongoVersion:"4.2.1", DownloadURL:"https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1804-4.2.1.tgz", MongodBin:"", Logger:(*log.Logger)(nil), LogLevel:2, StartupTimeout:10000000000}
[memongo] [INFO]  mongod from https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1804-4.2.1.tgz does not exist in cache, downloading to /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1804-4_2_1_tgz_5a13ca5024/mongod
[memongo] [INFO]  finished downloading mongod to /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1804-4_2_1_tgz_5a13ca5024/mongod in 40.671677188s
[memongo] [DEBUG] Using binary /home/runner/.cache/memongo/mongodb-linux-x86_64-ubuntu1804-4_2_1_tgz_5a13ca5024/mongod
[memongo] [DEBUG] Starting mongod
[memongo] [DEBUG] Started mongod; starting watcher
[memongo] [DEBUG] Started watcher; waiting for mongod to report port number
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.001+0000 I  CONTROL  [main] Automatically disabling TLS 1.0, to force-enable TLS 1.0 specify --sslDisabledProtocols 'none'
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.005+0000 I  CONTROL  [initandlisten] MongoDB starting : pid=3623 port=0 dbpath=/tmp/973013334 64-bit host=fv-az39-287
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.005+0000 I  CONTROL  [initandlisten] db version v4.2.1
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.005+0000 I  CONTROL  [initandlisten] git version: edf6d45851c0b9ee15548f0f847df141764a317e
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.005+0000 I  CONTROL  [initandlisten] OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.005+0000 I  CONTROL  [initandlisten] allocator: tcmalloc
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.005+0000 I  CONTROL  [initandlisten] modules: none
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.005+0000 I  CONTROL  [initandlisten] build environment:
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.005+0000 I  CONTROL  [initandlisten]     distmod: ubuntu1804
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.005+0000 I  CONTROL  [initandlisten]     distarch: x86_64
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.005+0000 I  CONTROL  [initandlisten]     target_arch: x86_64
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.005+0000 I  CONTROL  [initandlisten] options: { net: { port: 0 }, storage: { dbPath: "/tmp/973013334", engine: "ephemeralForTest" } }
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] ** WARNING: Access control is not enabled for the database.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] **          Read and write access to data and configuration is unrestricted.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] ** WARNING: This server is bound to localhost.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] **          Remote systems will be unable to connect to this server. 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] **          Start the server with --bind_ip <address> to specify which IP 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] **          addresses it should serve responses from, or with --bind_ip_all to
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] **          bind to all interfaces. If this behavior is desired, start the
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] **          server with --bind_ip 127.0.0.1 to disable this warning.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] ** WARNING: /sys/kernel/mm/transparent_hugepage/enabled is 'always'.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] **        We suggest setting it to 'never'
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] ** WARNING: soft rlimits too low. rlimits set to 27785 processes, 65536 files. Number of processes should be at least 32768 : 0.5 times number of files.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] ** NOTE: The ephemeralForTest storage engine is for testing only. 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] **       Do not use in production.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.008+0000 I  CONTROL  [initandlisten] 
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.011+0000 I  STORAGE  [initandlisten] createCollection: admin.system.version with provided UUID: d2b39e52-38ef-4f1a-8f5f-4abc1c600406 and options: { uuid: UUID("d2b39e52-38ef-4f1a-8f5f-4abc1c600406") }
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.011+0000 I  INDEX    [initandlisten] index build: done building index _id_ on ns admin.system.version
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.011+0000 I  SHARDING [initandlisten] Marking collection admin.system.version as collection version: <unsharded>
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.011+0000 I  COMMAND  [initandlisten] setting featureCompatibilityVersion to 4.2
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.011+0000 I  SHARDING [initandlisten] Marking collection local.system.replset as collection version: <unsharded>
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.011+0000 I  STORAGE  [initandlisten] Flow Control is enabled on this deployment.
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.011+0000 I  SHARDING [initandlisten] Marking collection admin.system.roles as collection version: <unsharded>
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.011+0000 I  STORAGE  [initandlisten] createCollection: local.startup_log with generated UUID: 8d63a061-bdb0-448c-97d0-a378dce8f2bb and options: { capped: true, size: 10485760 }
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.011+0000 I  INDEX    [initandlisten] index build: done building index _id_ on ns local.startup_log
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.011+0000 I  SHARDING [initandlisten] Marking collection local.startup_log as collection version: <unsharded>
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.011+0000 I  FTDC     [initandlisten] Initializing full-time diagnostic data capture with directory '/tmp/973013334/diagnostic.data'
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.012+0000 I  SHARDING [LogicalSessionCacheRefresh] Marking collection config.system.sessions as collection version: <unsharded>
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.012+0000 I  STORAGE  [LogicalSessionCacheRefresh] createCollection: config.system.sessions with provided UUID: e593da39-d8fc-437e-85c6-83f4c5a849d7 and options: { uuid: UUID("e593da39-d8fc-437e-85c6-83f4c5a849d7") }
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.012+0000 I  INDEX    [LogicalSessionCacheRefresh] index build: done building index _id_ on ns config.system.sessions
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.012+0000 I  INDEX    [LogicalSessionCacheRefresh] index build: starting on config.system.sessions properties: { v: 2, key: { lastUse: 1 }, name: "lsidTTLIndex", ns: "config.system.sessions", expireAfterSeconds: 1800 } using method: Hybrid
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.012+0000 I  INDEX    [LogicalSessionCacheRefresh] build may temporarily use up to 500 megabytes of RAM
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.012+0000 I  INDEX    [LogicalSessionCacheRefresh] index build: collection scan done. scanned 0 total records in 0 seconds
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.013+0000 I  INDEX    [LogicalSessionCacheRefresh] index build: inserted 0 keys from external sorter into index in 0 seconds
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.013+0000 I  INDEX    [LogicalSessionCacheRefresh] index build: done building index lsidTTLIndex on ns config.system.sessions
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.014+0000 I  NETWORK  [initandlisten] Listening on /tmp/mongodb-0.sock
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.014+0000 I  NETWORK  [initandlisten] Listening on 127.0.0.1
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.014+0000 I  NETWORK  [initandlisten] waiting for connections on port 37061
[memongo] [DEBUG] mongod started up and reported a port number after 30.055214ms
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.014+0000 I  SHARDING [LogicalSessionCacheReap] Marking collection config.transactions as collection version: <unsharded>
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.015+0000 I  NETWORK  [listener] connection accepted from 127.0.0.1:37286 #1 (1 connection now open)
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.017+0000 I  NETWORK  [listener] connection accepted from 127.0.0.1:37290 #2 (2 connections now open)
[memongo] [DEBUG] [Mongod stdout] 2021-07-20T21:33:43.017+0000 I  NETWORK  [conn2] received client metadata from 127.0.0.1:37290 conn2: { driver: { name: "mongo-go-driver", version: "v1.0.3" }, os: { type: "linux", architecture: "amd64" }, platform: "go1.13.15" }
--- FAIL: TestDefaultOptions (251.48s)
    --- FAIL: TestDefaultOptions/3.2.22 (53.94s)
        require.go:794: 
            	Error Trace:	main_test.go:23
            	Error:      	Received unexpected error:
            	            	timed out waiting for mongod to start
            	Test:       	TestDefaultOptions/3.2.22
    --- FAIL: TestDefaultOptions/3.4.21 (60.02s)
        require.go:794: 
            	Error Trace:	main_test.go:23
            	Error:      	Received unexpected error:
            	            	timed out waiting for mongod to start
            	Test:       	TestDefaultOptions/3.4.21
    --- FAIL: TestDefaultOptions/3.6.13 (58.37s)
        require.go:794: 
            	Error Trace:	main_test.go:23
            	Error:      	Received unexpected error:
            	            	timed out waiting for mongod to start
            	Test:       	TestDefaultOptions/3.6.13
FAIL
coverage: 63.9% of statements
FAIL	github.com/sbansal7/memongo	251.550s
ok  	github.com/sbansal7/memongo/memongolog	1.011s	coverage: 90.9% of statements
ok  	github.com/sbansal7/memongo/mongobin	40.482s	coverage: 90.5% of statements
ok  	github.com/sbansal7/memongo/monitor	1.020s	coverage: 83.3% of statements
FAIL
Error: Process completed with exit code 1. ```

**Comment by saniales** [comment]
This is interesting because I did not merge your changes and i am getting the same errors. Will dig into this, thanks 😊

**Comment by sbansal7** [comment]
I think it might be because of really old versions of mongo not being supported any more since it depends on libcrypto.so.1.0.0, which is not installed on ubuntu any more. Try updating supported mongo versions in tests.

**Comment by sbansal7** [comment]
See, this one worked with my latest commit

**Comment by saniales** [comment]
Closed by #3
